### ISSUE TITLE
[Feature Request]: "Give All", "Sell All", "Fence All", "Take All (pickpocketing), "Set Merchant Gold" options

### ISSUE THREAD

**Issue by Khamazandu** [Bradenm1/SkyrimMods#5]
Hey all, 

Let me just start off by saying thanks for all the hard work that's gone into this mod. It's wonderful, and has made Skyrim super fun for me. 

That said, I know that you guys just got done with a commit for Cheat Room and are getting ready to update said mod on all the various platforms (can't wait to try it out when that happens 😊), but I was wondering if you could add the ones listed in the title to a future commit?

The reason I'm asking for these in particular are for QoL. 

Basically, I made a rookie mistake and maxed out my character's metrics, so now I'm finding the occasional super chest with 100k lbs of items in them - 10 or so later and it takes my og Xbox One close to a minute to load the items menu and manipulate my inventory in any way. 

I know that I could use the "unequip-all" button in CR, but some of my items come from other mods and can be difficult or impossible to reacquire if I do so. 

I'd just like to be able to mass dump them into a no-limit CR chest that I could sort later. Maybe it could be a screen prompt when pressing "unequip-all" to just trash it like normal or else dump it into a chest(s). 

The rest are just to make bulk selling/thieving easier. It's not that I need money - CR makes that easy to get around - but I'd like to get some gold when I prune my inventory, otherwise it just feels like a waste. Hell, even just a cheat spell that allows me to convert non-equipped/quest item inventory to gold at its full value would be awesome. 

I don't know how easy all of that would be to achieve, but it would be awesome if you could.

Thanks!

**Comment by Bradenm1** [comment]
This mod 'https://bethesda.net/en/mods/skyrim/mod-detail/3247458' actually has the feature that stores your items. I'm not sure if it's still there, though.

I can look into these things but I'm not actively working on the mod.

**Comment by Khamazandu** [comment]
Hey there! 

Thanks for getting back to me.

Funny enough, I actually found and tried that mod out before making that initial request. 

It's Auto-Sort feature did actually manage to cut my inventory in half before it stopped working. 

As for the merchant in that mod, he only has 40k gold (though it does reset every time you exit and re-enter his shop dialog). And selling items one at a time is really slow due to how much I'm still carrying around (~350k lbs). 

I did try using another mod that triples merchant gold (which did bring him up to 120k) but that somehow broke the transaction system resulting in his numbers going down but mine staying the same.

Anyways, I didn't realize the project was inactive at this point. I just saw a whole bunch of items get added/modified since January, so I got hopeful.

Obviously, if you or anyone else that's been contributing to CR isn't interested then don't worry about it. 

This was just my attempt at trying to get somebody to help fix my mistakes, and maybe add some useful features in the process.

Thanks again for everything that you have done, it has been very much appreciated.

**Comment by gir489** [comment]
I tried to implement this while working on The Merchant. It's not possible to iterate a container's contents without using a stored array. I noticed that eventually when trying to move a lot of items in to a container for "sell all", it was slower than when I just sold them one at a time to the merchant and kept clearing his inventory to speed it back up again. This is just due to the slow nature of how the Creation Engine handles arrays. They (kinda) addressed this in Fallout 4, by moving to 64-bit and using 32-bit internal handlers, but the problem still exists if you get to nonsense amounts of stuff in your inventory.

TL;DR: What you see is what you get.

**Comment by Khamazandu** [comment]
Hey gir489!

Thanks for that insight. In that case, I'll close this issue. 

- Khamazandu
